### PR TITLE
Improve UfsIOBench and HdfsValidationTool

### PR DESCRIPTION
**Pull request by jiacheliu3**
This PR contains a few improvements to the HDFS tools:
1. Set process type of `UfsIOBench` as JOB_WORKER as it always runs in a job, invoked by a job worker.
1. Added description to each of the `TaskResult` from `HdfsValidationTool`. Each description corresponds to what the validation task did.
1. Removed `UfsContractTest` from `HdfsValidationTool`, as we only care about the accessibility of the UFS path, not all possible UFS operations.
1. Catch more exceptions now in the tools.

### CODE REVIEW

File: integration/tools/validation/src/main/java/alluxio/cli/ValidateEnv.java [hunk 1]
```diff
@@ -105,16 +105,19 @@ public ValidateEnv(String path, AlluxioConfiguration conf) {
     mConf = conf;
 
     // HDFS configuration validations
-    registerTask("ufs.hdfs.config.correctness", "validate HDFS configuration files",
+    registerTask("ufs.hdfs.config.correctness",
+            "This validates HDFS configuration files like core-site.xml and hdfs-site.xml.",
             new HdfsConfValidationTask(mPath, mConf), mCommonTasks);
     registerTask("ufs.hdfs.config.parity",
-            "validate HDFS-related configurations",
+            "If the Hadoop config directory is specified, this validates if that is "
```
Comment (LuQQiu) [inline integration/tools/validation/src/main/java/alluxio/cli/ValidateEnv.java:112]: ?? This validates if that is 
sounds a little unclear

File: integration/tools/validation/src/main/java/alluxio/cli/ValidateEnv.java [hunk 2]
```diff
@@ -105,16 +105,19 @@ public ValidateEnv(String path, AlluxioConfiguration conf) {
...
+                    + "consistent with the HDFS configuration given to Alluxio.",
             new HdfsConfParityValidationTask(mPath, mConf), mCommonTasks);
     registerTask("ufs.hdfs.config.proxyuser",
-            "validate proxyuser configuration in hdfs for alluxio",
+            "This validate proxy user configuration in HDFS for Alluxio. "
```
Comment (LuQQiu) [inline integration/tools/validation/src/main/java/alluxio/cli/ValidateEnv.java:116]: validates

Comment (LuQQiu) [reply integration/tools/validation/src/main/java/alluxio/cli/ValidateEnv.java:116]: Prefer to not use "this", or at least be clearer like "this task validates"

**Comment by madanadit** [comment]
```
Tests in error: 
alluxio.client.fs.FreeAndDeleteIntegrationTest.deleteDir(alluxio.client.fs.FreeAndDeleteIntegrationTest)
  Run 1: FreeAndDeleteIntegrationTest.deleteDir:145 » TestTimedOut test timed out after...
  Run 2: FreeAndDeleteIntegrationTest.deleteDir:145 » TestTimedOut test timed out after...

alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.completedEditLogDeletion(alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest)
  Run 1: UfsJournalIntegrationTest.completedEditLogDeletion:212->multiEditLogTestUtil:515->createFsMasterFromJournal:603 » Runtime
  Run 2: UfsJournalIntegrationTest.completedEditLogDeletion:212->multiEditLogTestUtil:515->createFsMasterFromJournal:603 » Runtime
```

**Comment by madanadit** [comment]
alluxio-bot, test this please.

**Comment by madanadit** [comment]
```
Tests in error: 
  TailCommandIntegrationTest.alluxio.client.cli.fs.command.TailCommandIntegrationTest » Runtime
```

**Comment by madanadit** [comment]
@ZacBlanco please take a look and confirm you are okay with cherry-pick to branch-2.3. i'll do that once the build passes.

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Comment by madanadit** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please.

**Pull request merged**